operation/image_cropping_test.go

```diff
@@ -0,0 +1,174 @@
+		t.Fatalf("Cropping area is not correct: %v", crop_area)
+	} else {
+		t.Logf("Cropping area: %v", crop_area)
+	}
+
+	// Compare images, pixel-by-pixel.
+	for x := 0; x < cropped_img.Image.Bounds().Dx(); x++ {
+		for y := 0; y < cropped_img.Image.Bounds().Dy(); y++ {
+
+			// Calculate the starting point of cropping area.
+			crop_start := image.Point{X: crop_area.Min.X, Y: crop_area.Min.Y}
+
+			// Calculate the current evaluating point on the original image by shifting the starting point of cropping area.
+			current_point_original_image := crop_start.Add(image.Point{X: x, Y: y})
+
+			// NOTE: This is a tricky part. If using At(x, y) to compare the pixel, it may not work as expected.
+			//       So, we use RGBA64At(x, y) instead.
+			if img.Image.Bounds().RGBA64At(current_point_original_image.X, current_point_original_image.Y) != cropped_img.Image.Bounds().RGBA64At(x, y) {
+				t.Logf("Image 1: %v", img.Image.At(x, y))
+				t.Logf("Image 2: %v", cropped_img.Image.At(x, y))
+				t.Fatalf("Images are not equal at (%d, %d)", x, y)
+			}
+		}
+	}
+
+}
+
+func TestImageCroppingCenter(t *testing.T) {
+
+	// Load image.
+	img, err := CreateImageFromFile("test_resources/test_ayaya.png")
+	if err != nil {
+		t.Fatalf("Failed to load image: %v", err)
+	}
+
+	img = img.Then(Decode())
+
+	cropped_img := img.Then(Crop(11, 15, "center"))
+
+	if cropped_img.LastError() != nil {
+		t.Fatalf("Failed to crop image: %v", cropped_img.LastError())
+	}
+
+	// Calculate the cropping area.
+	crop_area := CenterAlignment(img.Image.Bounds(), 11, 15)
+
+	// Check if cropped image bounds are correct.
+	// 14 - int(11 / 2) = 9
+	// 14 - int(15 / 2) = 7
+	if crop_area.Min.X != 9 || crop_area.Min.Y != 7 || crop_area.Max.X != 20 || crop_area.Max.Y != 22 {
+		t.Fatalf("Cropping area is not correct: %v", crop_area)
+	} else {
+		t.Logf("Cropping area: %v", crop_area)
+	}
+
+	// Compare images, pixel-by-pixel.
+	for x := 0; x < cropped_img.Image.Bounds().Dx(); x++ {
+		for y := 0; y < cropped_img.Image.Bounds().Dy(); y++ {
+
+			// Calculate the starting point of cropping area.
+			crop_start := image.Point{X: crop_area.Min.X, Y: crop_area.Min.Y}
+
+			// Calculate the current evaluating point on the original image by shifting the starting point of cropping area.
+			current_point_original_image := crop_start.Add(image.Point{X: x, Y: y})
+
+			// NOTE: This is a tricky part. If using At(x, y) to compare the pixel, it may not work as expected.
+			//       So, we use RGBA64At(x, y) instead.
+			if img.Image.Bounds().RGBA64At(current_point_original_image.X, current_point_original_image.Y) != cropped_img.Image.Bounds().RGBA64At(x, y) {
+				t.Logf("Image 1: %v", img.Image.At(x, y))
+				t.Logf("Image 2: %v", cropped_img.Image.At(x, y))
+				t.Fatalf("Images are not equal at (%d, %d)", x, y)
+			}
+		}
+	}
+
+}
+
+func TestInvalidBoundary(t *testing.T) {
+
+	// Load image.
+	img, err := CreateImageFromFile("test_resources/test_ayaya.png")
```

Comment:
Can we use `errors.Is` for error comparison to ensure we're correctly identifying the expected error type? :D

- [https://pkg.go.dev/errors](https://pkg.go.dev/errors)

Reply:
I'll do more research of this part.
Commonly, it just use simple compare to determine type of errors.
```go
if err == ErrSomethingNotWork {
    log.Fatelf("This not works.")
}
```